Editor/UI/Wizard/FirstTime/FirstTimeWizard.cs

```diff
@@ -4,14 +4,33 @@
 
 namespace Innoactive.CreatorEditor.UI.Wizard
 {
+    [InitializeOnLoad]
     internal static class FirstTimeWizard
     {
-        [MenuItem("Innoactive/Wizard")]
+        public const string skipWizardKey = "SkipWizard";
+
+        static FirstTimeWizard()
+        {
+            EditorApplication.update += ShowOnLoad;
+        }
+
+        private static void ShowOnLoad()
+        {
+            if (EditorPrefs.GetBool(skipWizardKey) == false && CourseAssetUtils.DoesAnyCourseExist() == false)
+            {
+                Show();
+            }
```

Comment:
Why not simply put this in the constructor?

Reply:
Because it's too early and the popup shows up before unity is displayed.

Reply:
What Simon said:
Because it needs to be triggered by the EditorApplication.update event. Otherwise it will really be done on InitializeOnLoad which happens while starting Unity and therefore kinda blocks the start of Unity due to the wizard being Modal. The update event also prevents the Wizard to block finishing compiling which is way better.
Unfortunately Unity did not provide any method up to today that is triggered as soon as Unity completely opened up :(